include/rpm/rpmkeyring.h

```diff
@@ -107,6 +107,22 @@ rpmPubkey rpmPubkeyLink(rpmPubkey key);
  */
 char * rpmPubkeyBase64(rpmPubkey key);
 
+/** \ingroup rpmkeyring
+ * Return fingerprint of primary key
+ * @param key		Pubkey
+ * @param fp		Fingerprint data
+ * @param fplen		Length of Fingerprint
+ * @return		0 on success
+ */
+int rpmPubkeyFingerprint(rpmPubkey key, uint8_t **fp, size_t *fplen);
+
+/** \ingroup rpmkeyring
+ * return primary key (or key itself iff it is the primary key)
+ * @param key           Pubkey
+ * @return		Primary key
+ */
+rpmPubkey rpmPubkeyMainKey(rpmPubkey key);
```

Comment:
Why "main" key and not "primary" key?

Reply:
Primary because I am unfamiliar with the nomenclature. If primary key is the right term we should rename that. `rpmPubkeyPrimarykey` or `rpmPrimaryPubkey`?

Reply:
> Primary because I am unfamiliar with the nomenclature. 

:D

> `rpmPubkeyPrimarykey` or `rpmPrimaryPubkey`?

`rpmPubkeyPrimarykey` sounds better to me.

Reply:
OK, turns out `mainkey` was already used as a parameter name for the subkey parsing functions. Renamed it there, too.